src/builder/commands/composer.rs

```diff
@@ -0,0 +1,269 @@
+
+    if ctx.composer_settings.install_runtime {
+        try!(setup_include_path(ctx));
+    }
+
+    Ok(())
+}
+
+fn setup_include_path(ctx: &mut Context) -> Result<(), String> {
+    let vagga_ini_content = {
+        let include_path = ctx.composer_settings
+            .include_path.clone()
+            .unwrap_or(DEFAULT_INCLUDE_PATH.to_owned());
+        format!("include_path={}", include_path)
+    };
+
+    // find conf.d directories (various can exist on Ubuntu)
+    let conf_dirs = try!(find_dirs(ctx, "/etc/php**/conf.d"));
+
+    for conf_d in conf_dirs.iter() {
+        // create vagga.ini file
+        try!(create_vagga_ini(&conf_d.join("vagga.ini"), &vagga_ini_content));
+    }
+
+    if !conf_dirs.is_empty() {
+        // vagga.ini file(s) created, we're done here
+        return Ok(())
+    }
+
+    // If we didn't find any conf.d, ask 'php --ini'
+    let conf_d = try!(ask_php_for_conf_d(ctx));
+
+    // create conf.d
+    if !conf_d.exists() {
+        try!(file_util::create_dir(&conf_d, true)
+        .map_err(|e| format!("Error creating directory {:?}: {}", conf_d, e)));
+    }
+    // create vagga.ini file
+    try!(create_vagga_ini(&conf_d, &vagga_ini_content));
+
+    Ok(())
+}
+
+fn create_vagga_ini(location: &Path, content: &str) -> Result<(), String> {
+    File::create(location)
+        .and_then(|mut f| f.write_all(content.as_bytes()))
+        .map_err(|e| format!("Error creating file {:?}: {}", location, e))
+}
+
+fn find_dirs(ctx: &mut Context, search: &str) -> Result<Vec<PathBuf>, String> {
+    let args = [
+        "find".to_owned(),
+        "/etc".to_owned(),
+        "-path".to_owned(),
+        search.to_owned(),
+    ];
+    let dirs: Vec<PathBuf> = try!(capture_command(ctx, &args, &[])
+        .and_then(|x| String::from_utf8(x) // convert output to String
+            .map(|x| x.lines() // iterate over lines
+                .filter(|l| !l.trim().is_empty()) // filter empty lines
+                .map(|l| PathBuf::from(format!("/vagga/root{}", l))) // vagga/root{/etc/php}
+                .collect())
+            .map_err(|e| format!("Error parsing command output: {}", e)))
+        .map_err(|e| format!("Error reading command output: {}", e)));
+    Ok(dirs)
+}
+
+fn ask_php_for_conf_d(ctx: &mut Context) -> Result<PathBuf, String> {
+    let runtime_exe = ctx.composer_settings
+        .runtime_exe
+        .clone()
+        .unwrap_or(DEFAULT_RUNTIME.to_owned());
+
+    let args = [runtime_exe, "--ini".to_owned()];
+    let output = try!(capture_command(ctx, &args, &[])
+        .and_then(|x| String::from_utf8(x)
+            .map_err(|e| format!("Error parsing command output: {}", e)))
+        .map_err(|e| format!("Error reading command output: {}", e)));
+
+    // match any line that ends with /etc/php*/**/conf.d, get first result
+    let re = try_msg!(Regex::new(r#"(?m).*?(/etc/php\d/.*?conf.d)$"#), "Invalid regex: {err}");
```

Comment:
this should use `.expect("invalid regex")` rather than `try_msg`. try_msg is for runtime errors. But regex compilation will fail only when you have put wrong regex (which will be caught by unit tests)